### PR TITLE
Fix notify sighup tests

### PR DESCRIPTION
**Pull request by SimonRichardson**
The `SIGHUP` tests fail because the tests grabbed the process of the current test, but that might not be the right thread. We could use runtime lock to thread, but that doesn't seem ideal. Instead, we move the notify logic out of the worker and allow a test one to inject.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

```sh
$ go test -v ./worker/controlleragentconfig  -check.v -check.f="workerSuite" -count=100 -race
```

## Links

**Jira card:** JUJU-4675

### CODE REVIEW

File: worker/controlleragentconfig/manifold.go [hunk 1]
```diff
@@ -54,3 +60,12 @@ func Manifold(config ManifoldConfig) dependency.Manifold {
 func configOutput(in worker.Worker, out any) error {
 	return nil
 }
+
+// Notify sets up the signal handler for the worker.
+func Notify(ctx context.Context, ch chan os.Signal) {
+	if ctx.Err() != nil {
```
Comment (nvinuesa) [inline worker/controlleragentconfig/manifold.go:66]: this `ctx` is only needed to check if it's in error? shouldn't we check for `Done()` as well?

Comment (SimonRichardson) [reply worker/controlleragentconfig/manifold.go:66]: `Err()` is a proxy for done in this case.

See the documentation:

```
	// If Done is not yet closed, Err returns nil.
	// If Done is closed, Err returns a non-nil error explaining why:
	// Canceled if the context was canceled
	// or DeadlineExceeded if the context's deadline passed.
	// After Err returns a non-nil error, successive calls to Err return the same error.
```

Comment (nvinuesa) [reply worker/controlleragentconfig/manifold.go:66]: oh right! thanks

**Review by nvinuesa — APPROVED** [review]
LGTM

**Pull request merged**